Dockerfile-dev

```diff
@@ -4,7 +4,7 @@ LABEL maintainer "devs@bigchaindb.com"
 
 RUN apt-get update \
     && apt-get install -y vim \
-    && pip install -U pip \
+    && pip install -U pip==18.1 \
```

Comment:
Maybe we could just remove the `--process-dependency-links` option below (currently in line 35) instead?

Reply:
I did a search for `dependency` in this repo and it looks like that `--process-dependency-links` option shows up in many other places as well.

Maybe all of them could be removed, _and_ we could require `pip>=18.1`?

Reply:
I agree, to remove the flag is a better option if it is not used, on my side I wasn't sure if there are  non-PyPi packages during installation that may be affected by removing  `--process-dependency-link`.